api/src/pcapi/core/offers/api.py

```diff
@@ -1824,3 +1826,67 @@ def update_used_stock_price(stock: models.Stock, new_price: float) -> None:
             event=first_finance_event,
             reason=finance_models.PricingLogReason.CHANGE_AMOUNT,
         )
+
+
+def _is_allocine(provider_id: int) -> bool:
+    allocine_products_provider_id = get_allocine_products_provider_id()
+    allocine_stocks_provider_id = get_provider_by_local_class("AllocineStocks").id
+    return provider_id in (allocine_products_provider_id, allocine_stocks_provider_id)
+
+
+def _update_product_extra_data(product: offers_models.Product, movie: offers_models.Movie) -> None:
+    if not product.extraData:
+        product.extraData = offers_models.OfferExtraData()
+
+    extra_data = movie.extra_data or offers_models.OfferExtraData()
+    if movie.allocine_id:
+        extra_data["allocineId"] = int(movie.allocine_id)
+    if movie.visa:
+        extra_data["visa"] = movie.visa
+
+    product.extraData.update((key, value) for key, value in extra_data.items() if value is not None)  # type: ignore[typeddict-item]
+
+
+def _update_movie_product(
+    product: offers_models.Product, movie: offers_models.Movie, provider_id: int, id_at_providers: str
+) -> None:
+    product.description = movie.description
+    product.durationMinutes = movie.duration
+    product.idAtProviders = id_at_providers
+    product.lastProviderId = provider_id
+    product.name = movie.title
+    _update_product_extra_data(product, movie)
+
+
+def upsert_movie_product_from_provider(
+    movie: offers_models.Movie, provider_id: int, id_at_providers: str
+) -> offers_models.Product | None:
+    if not movie.allocine_id and not movie.visa:
+        logger.warning("Cannot create a movie product without allocineId nor visa")
+        return None
+
+    existing_product = None
+    if movie.allocine_id:
+        existing_product = offers_repository.get_movie_product_by_allocine_id(movie.allocine_id)
+    if not existing_product and movie.visa:
+        existing_product = existing_product or offers_repository.get_movie_product_by_visa(movie.visa)
```

Comment:
Vu qu'il y a la condition `not existing_product` au dessus
```suggestion
        existing_product = offers_repository.get_movie_product_by_visa(movie.visa)
```